src/cr/cube/dimension.py

```diff
@@ -982,6 +982,35 @@ def translate_element_id(self, _id) -> Optional[str]:
             return _id
         if _id in self._raw_element_ids:
             return self._subvar_aliases[self._raw_element_ids.index(_id)]
+        # --- In case of MR dimensions with HS there's an edge case where the hidden
+        # --- column refers to the subvariable idx position defined in the raw
+        # --- elements elements ids.
+        # --- For example id = "2",
+        # --- subvar_ids = ('savory+spicy','1','spicy+sweet','2','savory+sweet','3')
+        # --- subvar_aliases = ('savory___spicy_at_top',
+        #                       'enjoy_mr_savory__4',
+        #                       'spicy___sweet_after_enjoy_mr_savory__4',
+        #                       'enjoy_mr_spicy__4',
+        #                       'savory___sweet_after_enjoy_mr_spicy__4',
+        #                       'enjoy_mr_sweet__4')
+        # --- Relying on the subvar_ids this will get the self._subvar_ids.index(_id)
+        # --- that corresponds to 3 and so subvar_aliases[3] == enjoy_mr_spicy__4.
+        # --- This means that we're hiding the col Spicy rather than Savory.
+        # --- Using the _raw_element_ids instead will return the correct hidden alias.
+        # --- NOTE: this goes away when we will use aliases rather than id.
+        if self._has_mr_insertion:
+            elements = self._dimension_dict["type"]["elements"]
+            insertions_mask = [
+                1 if el["value"].get("references", {}).get("anchor") else 0
```

Comment:
I'd rather use the construction `"anchor" in obj` than `obj.get("anchor")`. The first one is more explicit in that you don't care about the contents of the anchor, but just that it's present (because it defines the insertion).

Reply:
right thx